src/librustc_resolve/diagnostics.rs

```diff
@@ -16,6 +16,8 @@
 register_long_diagnostics! {
 
 E0154: r##"
+## WARNING: Unused error code
```

Comment:
Perhaps this can be clearer about what it means to be unused. As a warning this sounds like the user has done something wrong. What about something like 'Note: this error code is no longer emitted by the compiler.'?